docs/proposals/apiman-integration.md

```diff
@@ -0,0 +1,123 @@
+* **UC05** As a cluster administrator, I want to deploy APIMan and its components reusing existing infrastructure components where possible (e.g. ElasticSearch), so I can minimize infrastructure components.
+* **UC06** As a cluster administrator, I want API management interface to be able to discover the route to Kibana, so it can be linked to from the management interface. 
+* **UC07** As an API consumer, I want to configure my services to consume other services
+
+## Specification
+
+API services will be scoped to OpenShift projects. Projects will have one-to-one relationship to an APIMan namespace (formally organization). APIMan will utilize namespace
+to manage policy regarding the service.
+ 
+### Service Annotations
+Services intended to be managed by APIMan and exposed as API Endpoints will be annotated<sup>[5](#r5)</sup>.  The annotations are repeated below for convenience:
+
+```
+  apiVersion: "v1"
+  kind: "Service"
+  metadata: 
+    annotations: 
+      api.service.kubernetes.io/protocol: REST
+      api.service.kubernetes.io/scheme: http
+      api.service.kubernetes.io/path: cxfcdi
+      api.service.kubernetes.io/description-path: cxfcdi/swagger.json
+      api.service.kubernetes.io/description-language: SwaggerJSON
+```
+
+### Origin Web Console UI extension
+A custom extension<sup>[6](#r6)</sup> will be created and hooked into the web console to support API endpoints.  The extension
+will allow a user to navigate to the APIMan management interface to:
+* Manage an API endpoint
+* Expose a service as a new API endpoint
+
+The extension will provide the following details to the APIMan gateway:
+* Service name
+* Service namespace
+* User's Oauth token
+
+Calls from the extension to the APIMan management interface will utilize a REST POST call where the oauth token is part of the payload.  It will
+utilize a similiar design<sup>[7](#r7)</sup> that is realized by the OpenShift origin aggregated logging integration and the auth proxy<sup>[8](#r8)</sup>.  It is necessary for
+security reasons to not add the token as a query parameter.  
+
+### API Services Gateway
+The APIMan gateway will be deployed as a cluster-wide infrastructure component to handle services exposed as API endpoints.  Internal and external clients can use the gateway to access APIs.  Admins should be able to configure the cluster to limit direct access to services and encourage consumers of API endpoints to utilize the gateway.
+
+<s>APIman will be deployed as an additional gateway router that internal and external clients can use to access APIs. It is not an exclusive router, but admins should be able to configure the cluster to limit access to services directly and instead encourage applications bouncing off the gateway. The gateway would act as a 'router', although it will effectively be looking at annotated services directly.</s>
+
+**Need: Document how a cluster admin might setup their cluster to control services as alluded to in this section **
```

Comment:
To prevent pods in other projects from using services directly, the multitenant SDN plugin can be used.

There's no way I know of the prevent intra-project use of services - it's assumed you have control of your own project.

Since it's implemented via a router, this implies we're accessing services via routes instead of the service name. It would be really nice if the FQDN service names routed to the router directly though, instead of just failing and requiring a route.